libs/community/langchain_community/document_loaders/unstructured.py

```diff
@@ -255,44 +212,108 @@ class UnstructuredAPIFileLoader(UnstructuredFileLoader):
+                else:
+                    # If exists, append to text and update the metadata
+                    text_dict[page_number] += str(element.get("text")) + "\n\n"
+                    meta_dict[page_number].update(metadata)
+
+            # Convert the dict to a list of Document objects
+            for key in text_dict.keys():
+                yield Document(page_content=text_dict[key], metadata=meta_dict[key])
+        elif self.mode == "single":
+            metadata = self._get_metadata()
+            text = "\n\n".join([el.get("text") for el in elements_json])
+            yield Document(page_content=text, metadata=metadata)
+        else:
+            raise ValueError(f"mode of {self.mode} not supported.")
+
+    def _get_elements(self) -> list:
+        if isinstance(self.file_path, list):
+            elements = []
+            for path in self.file_path:
+                elements.extend(
+                    get_elements_from_api(
+                        file_path=path,
+                        api_key=self.api_key,
+                        api_url=self.url,
+                        **self.unstructured_kwargs,
+                    )
+                )
+            return elements
 
-    def _get_elements(self) -> List:
         return get_elements_from_api(
             file_path=self.file_path,
             api_key=self.api_key,
             api_url=self.url,
             **self.unstructured_kwargs,
         )
+    
+    def _get_metadata(self) -> dict:
+        return {"source": self.file_path}
+
+    def _post_process_elements(self, elements: list[dict]) -> list:
+        """Apply post processing functions to extracted unstructured elements.
 
+        Post processing functions are str -> str callables passed
+        in using the post_processors kwarg when the loader is instantiated.
+        """
+        for element in elements:
+            for post_processor in self.post_processors:
+                element["text"] = post_processor(element.get("text"))
+        return elements
+    
 
 class UnstructuredFileIOLoader(UnstructuredBaseLoader):
-    """Load files using `Unstructured`.
+    """Load file-like objects opened in read mode using `Unstructured`.
 
-    The file loader
-    uses the unstructured partition function and will automatically detect the file
-    type. You can run the loader in one of two modes: "single" and "elements".
-    If you use "single" mode, the document will be returned as a single
-    langchain Document object. If you use "elements" mode, the unstructured
-    library will split the document into elements such as Title and NarrativeText.
-    You can pass in additional unstructured kwargs after mode to apply
-    different unstructured settings.
+    The file loader uses the unstructured partition function and will automatically detect the file
+    type. You can run the loader in one of two modes: "single" and "elements". If you use "single"
```

Comment:
@MthwRobinson  The Loader also supports a "paged" mode. Should the docstrings be updated to mention this third mode or do we want to continue leaving it out and reference the unstructured's chunking options, which includes chunking `by_page`?